scout/models/phenotype_term.py

```diff
@@ -1,41 +1,21 @@
-# -*- coding: utf-8 -*-
+from typing import List, Optional
 
+from pydantic import BaseModel, validator
 
-# Hpo terms represents data from the hpo web
-class HpoTerm(dict):
-    """Represents a hpo term
-
-    _id = str, # Same as hpo_id
-    hpo_id = str, # Required
-    hpo_number = int, # Required
-    description = str,
-    genes = list, # List with integers that are hgnc_ids
-    ancestors = list # list with direct ancestors
-    all_ancestors = list # list with all ancestors ancestors in the whole tree
-    children = list
 
-    """
+# Hpo terms represents data from the hpo web
+class HpoTerm(BaseModel):
+    hpo_id: str
+    hpo_number: Optional[int]
+    description: str
+    ancestors: List = []
+    all_ancestors: List = []
+    children: List = []
+    genes: List = []
 
-    def __init__(
-        self,
-        hpo_id,
-        description,
-        genes=None,
-        ancestors=None,
-        all_ancestors=None,
-        children=None,
-    ):
-        super(HpoTerm, self).__init__()
-        self["_id"] = hpo_id
-        self["hpo_id"] = hpo_id
-        self["hpo_number"] = int(hpo_id.split(":")[-1])
-        self["description"] = description
-        # These are the direct ancestors
-        self["ancestors"] = ancestors or []
-        # This all ancestors of all ancestors
-        self["all_ancestors"] = all_ancestors or []
-        self["children"] = children or []
-        self["genes"] = genes or []
+    @validator("hpo_number", always=True)
+    def get_hpo_number(cls, value, values) -> int:
```

Comment:
Hm, I kind of agree with vulture this is a rather weird syntax when you dont really expect a value to be present. I don't know if it would accept `_` for an unused value (but probably renaming to like `unset_value` or sth. It's working, and would presumably look a little clearer in Pydantic2 so nothing to fret about.